src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -407,6 +408,28 @@ private void createMesosTask(Offer offer, Request request) {
     LOGGER.info("Launching task " + taskId.getValue() + " with URI " +
                 joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
 
+    for (final Computer computer : Jenkins.getInstance().getComputers()) {
```

Comment:
Actually we should remove pending delete if it's done launching. That's in the code now, a change was made to prevent a slave from starting and then deleted mid-test